Rl 1

Squidtito opened the pull request:
----Added player
Able to jump now
able to collide with surfaces

https://trell.ooo/workspace47241247/RL-1

Squidtito committed:
Player movement test

Sunkist

Squidtito committed:
A more stable player script

Added jump animation

Added jump

-Sunkist

keri4141 commented on scripts/player.gd:
```diff
@@ -0,0 +1,39 @@
+extends KinematicBody2D
+var vect = Vector2(0,0)
+var up = Vector2(0,-1)
```
can we give these variables types so that others can know what this variable type is

var up:Vector = Vector2(0,-1)

keri4141 commented on scripts/player.gd:
```diff
@@ -0,0 +1,39 @@
+extends KinematicBody2D
+var vect = Vector2(0,0)
```
add type here as well for Vector

keri4141 replied:
rename to velocity please

keri4141 commented on scripts/player.gd:
```diff
@@ -0,0 +1,39 @@
+# Called every frame. 'delta' is the elapsed time since the previous frame.
+func _physics_process(delta):
+		
+	if Input.is_action_pressed('d'):
```
rename these to the actual name of the direction ur moving at for the key mapping

keri4141 commented on scripts/player.gd:
```diff
@@ -0,0 +1,39 @@
+	if Input.is_action_pressed('d'):
+		vect.x = 200
+		$AnimatedSprite.set_flip_h(false)
+		$AnimatedSprite.offset = Vector2(0,0)
```
let's have this as
var rightOffset:Vector = Vector2(0,0)
$AnimatedSprite.offset = rightOffset

Squidtito committed:
Corrected inefficient code

Centered player sprites to remove offset code

properly defined new variables

Renamed variable "vect" to "velocity" of it being misunderstood

Renamed inputs into left, right, etc, instead of w, a, s, d.

Squidtito committed:
Delete Player.pdn

e
